File: 0131-palindrome-partitioning/0131-palindrome-partitioning.py
class Solution:
    def partition(self, s: str) -> List[List[str]]:
        
        def part_it(start = 0, parts = []):
            if(start >= len(s)):
                ans.append([''.join(ss) for ss in parts])
                return
            
            temp = []
            for i in range(start, len(s)):
                temp.append(s[i])
                if(temp == temp[::-1]):
                    part_it(i + 1, parts + [temp])
        
        
        ans = []
        part_it()
        return ans
        
                

File: 0131-palindrome-partitioning/0131-palindrome-partitioning.go
func partition(s string) [][]string {
    var res = [][]string{}
    var dp = [][]bool{}
    var size = len(s)
    var palinParts func(index int, currStack []string)

    for range size {
        var temp = []bool{}
        for range size {
            temp = append(temp, false)
        }
        dp = append(dp, temp)
    }
    
    palinParts = func(index int, currStack []string) {
        if index >= len(s) {
            var temp = make([]string, len(currStack))
            copy(temp, currStack)
            res = append(res, temp)
            return
        }
        var str = ""
        for i := index; i < len(s); i++ {
            str += string(s[i])
            if str[0] == s[i] && (i - index <= 2 || dp[index + 1][i - 1]) {
                dp[index][i] = true
                palinParts(i + 1, append(currStack, str))
            }
        }
    }
    
    palinParts(0, []string{})
    return res
}